src/services/web3/liquidity/liquidity.ts

```diff
@@ -160,15 +160,25 @@ export const removeLiquidity = async (
       poolToken.poolDecimals
     );
 
-    const tx = await contract.removeLiquidity(
-      expandToken(poolToken.amount, poolToken.poolDecimals),
-      [poolToken.bnt.token.address, poolToken.tkn.token.address],
-      [minBntReturn, minTknReturn]
-    );
-    onHash(tx.hash);
-
-    await tx.wait();
-    onCompleted();
+    if (poolToken.version < 28) {
```

Comment:
suggestion: if the only difference in the flow is how we liquidate, as in:
```
contract.liquidate(
  expandToken(poolToken.amount, poolToken.poolDecimals)
);
```
vs
```
contract.removeLiquidity(
  expandToken(poolToken.amount, poolToken.poolDecimals),
  [poolToken.bnt.token.address, poolToken.tkn.token.address],
  [minBntReturn, minTknReturn]
 );
```
then we can just do:
```
const tx = await liquidateContract(contract, poolToken, minBntReturn, minTknReturn);
```

where `liquidateContract` is a helper function that does the version check and calls the proper function

Reply:
fixed